cmd/thanos/query.go

```diff
@@ -426,9 +443,11 @@ func runQuery(
 
 		g.Add(func() error {
 			level.Info(logger).Log("msg", "Listening for StoreAPI gRPC", "address", grpcBindAddr)
+			querierIsReady = true
```

Comment:
There is slight time between gRPC serving request and setting readiness, but I think it's fine.. not regression for sure as we set readiness on `/metrics` page.